opm/input/eclipse/Schedule/SummaryState.hpp

```diff
@@ -122,6 +125,39 @@ class SummaryState {
     std::size_t num_wells() const;
     std::size_t size() const;
     bool operator==(const SummaryState& other) const;
+
+    template<class Serializer>
+    void serializeOp(Serializer& serializer)
+    {
+      serializer(sim_start);
+      serializer(elapsed);
+      serializer.map(values);
+      serializer.map(well_values);
+      serializer.set(m_wells);
+      serializer(well_names);
+      serializer.map(group_values);
+      serializer.set(m_groups);
+      serializer(group_names);
+      serializer.map(conn_values);
+    }
+
+    static SummaryState serializeObject()
+    {
+        auto st = SummaryState{TimeService::from_time_t(101)};
+
+        st.elapsed = 1.0;
+        st.values = {{"test1", 2.0}};
+        st.well_values = {{"test2", {{"test3", 3.0}}}};
+        st.m_wells = {"test4"};
+        st.well_names = {"test5"};
+        st.group_values = {{"test6", {{"test7", 4.0}}}},
+        st.m_groups = {"test7"};
+        st.group_names = {"test8"},
+        st.conn_values = {{"test9", {{"test10", {{5, 6.0}}}}}};
```

Comment:
This is probably okay as a way of testing the serialization code, but please be aware that this setup does not honour the internal invariants of the data members.  In other words, the object returned from `serializeObject()` here is not a valid `SummaryState`.

Reply:
yeah, i'm aware, but as it's only for the purpose of testing serialization i don't consider it a problem.

Reply:
> as it's only for the purpose of testing serialization i don't consider it a problem.

Right.  Let's hope no-one chooses to call the function outside of the testing code.